### PR TITLE
Update Imports from Blanket to Selective Import for Improved Code Maintainability

### LINKED ISSUE #5970: Use selective imports
https://github.com/Rdatatable/data.table/blob/07fd933e7d724d00d292b325da2cb5d79ecbcb77/NAMESPACE#L133

In general it's preferable (perhaps just IMO) to import selectively vs. blanket `import(pkg)`. Blanket imports risk unintended collisions, especially as packages evolve (add+remove functions), though the risk for the case of default packages (methods/utils/stats) is more borderline.

### PR DESCRIPTION
**Pull request by Nj221102**
**Summary:**
This pull request updates the imports in the codebase to use selective imports instead of blanket imports for better maintainability and reduced risk of conflicts.

**Changes Made:**
- Replaced `import(methods)` with `importFrom("methods", "S3Part<-", "slotNames")`.
- Replaced `import(utils)` with `importFrom("utils", "tail", "head", "packageVersion", "capture.output", "contrib.url", "download.file", "flush.console", "getS3method", "untar", "unzip")`.
- Replaced `import(stats)` with `importFrom("stats", "na.omit", "as.formula", "setNames", "terms", "time")`.

This PR Closes #5970

### CODE REVIEW

**Comment by Nj221102** [comment]
@MichaelChirico  plz checkout this PR if you have time :)

**Comment by MichaelChirico** [comment]
thanks! please remove quotes (`"`) wherever possible

**Comment by Nj221102** [comment]
> thanks! please remove quotes (`"`) wherever possible

removed quotes from everywhere except "S3Part<-" because it was causing some error.

**Comment by MichaelChirico** [comment]
> I removed quotes from everywhere except "S3Part<-" because it was causing some error.

right, NAMESPACE needs to parse as R, so `S3Part<-` needs to be quoted. Any "normal" R name (`?make.names`) needn't be quoted.

File: NAMESPACE [hunk 1]
```diff
@@ -130,11 +130,11 @@ S3method(melt, default)
 # and many packges on CRAN call dcast.data.table() and/or melt.data.table() directly. See #3082.
 export(melt.data.table, dcast.data.table)
 
-import(utils)
+importFrom(utils, tail, head, packageVersion, capture.output, contrib.url, download.file, flush.console, getS3method, untar, unzip)
```
Comment (MichaelChirico) [inline NAMESPACE:133]:
```suggestion
importFrom(utils, capture.output, contrib.url, download.file, flush.console, getS3method, head, packageVersion, tail, untar, unzip)
```

File: NAMESPACE [hunk 2]
```diff
@@ -130,11 +130,11 @@ S3method(melt, default)
...
 export(update_dev_pkg)
 S3method(tail, data.table)
 S3method(head, data.table)
-import(stats)
+importFrom(stats, na.omit, as.formula, setNames, terms, time)
```
Comment (MichaelChirico) [inline NAMESPACE:137]:
```suggestion
importFrom(stats, as.formula, na.omit, setNames, terms, time)
```

File: NAMESPACE [hunk 3]
```diff
@@ -130,11 +130,11 @@ S3method(melt, default)
 # and many packges on CRAN call dcast.data.table() and/or melt.data.table() directly. See #3082.
 export(melt.data.table, dcast.data.table)
 
-import(utils)
+importFrom(utils, capture.output, contrib.url, download.file, flush.console, getS3method, head, packageVersion, tail, untar, unzip)
 export(update_dev_pkg)
 S3method(tail, data.table)
 S3method(head, data.table)
-import(stats)
+importFrom(stats, as.formula, na.omit, setNames, terms, time)
```
Comment (MichaelChirico) [inline NAMESPACE:137]: I think `time` is only used as NSE, we don't actually need to import `stats::time`:

https://github.com/Rdatatable/data.table/blob/815e054495d787f4a258fc3c6201a11f13e610e2/R/test.data.table.R#L197

Comment (MichaelChirico) [reply NAMESPACE:137]: Instead, edit this line to define `time` as `NULL` as well:

https://github.com/Rdatatable/data.table/blob/815e054495d787f4a258fc3c6201a11f13e610e2/R/test.data.table.R#L194

That mimics what's done in the same file:

https://github.com/Rdatatable/data.table/blob/815e054495d787f4a258fc3c6201a11f13e610e2/R/test.data.table.R#L282

Comment (Nj221102) [reply NAMESPACE]: Done :)

**Comment by MichaelChirico** [comment]
Sorry, one more change -- please add yourself to `Authors@R` field for this change, since it is not just a "documentation" change.

LGTM otherwise, thanks!

**Comment by Nj221102** [comment]
> Sorry, one more change -- please add yourself to `Authors@R` field for this change, since it is not just a "documentation" change.
> 
> LGTM otherwise, thanks!

Done 👍

**Review by MichaelChirico — APPROVED** [review]

**Comment by MichaelChirico** [comment]
Thanks for the PR! You've been added as a project member to facilitate future collaboration! Check your e-mail. You'll be able to create branches directly from this repo going forward. Thanks again!

**Pull request merged**